ray-operator/controllers/ray/common/service_test.go

```diff
@@ -124,6 +126,33 @@ var (
 			},
 		},
 	}
+	instanceWithoutSvc = &rayv1.RayCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "raycluster-sample-nosvc",
+			Namespace: "default",
+		},
+		Spec: rayv1.RayClusterSpec{
+			HeadServiceAnnotations: map[string]string{
+				headServiceAnnotationKey1: headServiceAnnotationValue1,
+				headServiceAnnotationKey2: headServiceAnnotationValue2,
+			},
+			HeadGroupSpec: rayv1.HeadGroupSpec{
+				ServiceType: corev1.ServiceTypeClusterIP,
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "ray-head",
+								Ports: []corev1.ContainerPort{
+									{ContainerPort: 10001, Name: "client"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Why not use `instanceForSvc` directly? They are almost the same.

Reply:
They are different, instanceWithoutSvc does not include the "serve" container port for the service.